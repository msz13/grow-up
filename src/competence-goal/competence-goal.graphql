scalar Date

type Query {
    competenceGoals(status: GoalStatus, competence: String): [CompetenceGoal]
}

type Mutation {
    createCompGoal(competenceGoalInput: CompetenceGoalInput): CompetenceGoal
   # goalStatusToHold(goal_Id: ID): Boolean
   # goalStatusToDone(goal_Id: ID): Boolean
   # goalStatusToActive(goal_Id: ID): Boolean
   # incPerf(compGoal_Id: ID, fDayOfWeek: GraphQLDate, dayOfWeek: Int): Int
   # decPerf(compGoal_Id: ID, fDayOfWeek: GraphQLDate, dayOfWeek: Int): Boolean
    deleteCompGoal(goal_Id: ID): Boolean
    updateComGoal(goal_Id: ID, updateCompetenceGoalInput: UpdateCompetenceGoalInput): CompetenceGoal
}
  
type CompetenceGoal {
  id: String
  name: String! 
  competence: String
  target: Int #daily target of goal
  status: GoalStatus
  performance: ActiveGoalPerf  
  perfHistory: [GoalPerfHistory]
 }

 enum GoalStatus{
    ACTIVE
    DONE
    HOLD
 }



 type GoalDayPerf {
     date: Date
     perfCount: Int
     targetIsDone: Boolean
 }

 interface GoalPerf {
     # date of start active status
     startActive: Date
     #number of days of peforming goal
     dayCount: Int
     # score of perf
     overallPerf: Int
     weekCount: Int
     goalEffectivenes: Int
 }

 type ActiveGoalPerf implements GoalPerf {
     # date of start active status
     startActive: Date
     #number of days of peforming goal
     dayCount: Int
     # score of perf
     overallPerf: Int #czy potrzebne?
     weekCount: Int
     goalEffectivenes: Int
     goalDayPerf(from: Date, to: Date): [GoalDayPerf!]
 }

 type GoalPerfHistory implements GoalPerf {
     # date of start active status
     startActive: Date
     #date od end active status
     endActive: Date
     #number of days of peforming goal
     dayCount: Int
     # score of perf
     overallPerf: Int #czy wywaliÄ‡?
     weekCount: Int
     goalEffectivenes: Int
 }


 input CompetenceGoalInput {
     name: String!
     competence: String!
     target: Int!
     }


input UpdateCompetenceGoalInput {
     name: String
     competence: String
     target: Int
     }



