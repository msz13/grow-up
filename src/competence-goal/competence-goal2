  
interface CompetenceGoal {
  id: ObjectID
  name: String! 
  competence: String
  target: Int #daily target of goal
  status: GoalStatus
  perfHistory: [GoalPerfHistory]
 }

 enum GoalStatus{
    ACTIVE
    DONE
    HOLD
 }



 type GoalDayPerf {
     id: ID!
     date: Date
     perfCount: Int
     targetIsDone: Boolean
 }

type ActiveCompetenceGoal implements CompetenceGoal {
    id: ObjectID
    name: String! 
    competence: String
    target: Int #daily target of goal
    status: GoalStatus      
    perfHistory: [GoalPerfHistory] 
    
    # date of start active status
    startActive: Date
    #number of days of peforming goal
    dayCount: Int
    weekCount: Int
    # score of perf
    overallPerf: Int #czy potrzebne?
    goalEffectivenes: Int #number of days on target
    goalDaysPerf(from: Date, to: Date): [GoalDayPerf!]
       
}

 type PassiveCompetenceGoal {
    id: ObjectID
    name: String! 
    competence: String
    target: Int #daily target of goal
    status: GoalStatus
    perfHistory: [GoalPerfHistory]

 }


 type GoalPerfHistory implements GoalPerf {
     
    
     # date of start active status
     startActive: Date
     #date od end active status
     endActive: Date
     #number of days of peforming goal
     dayCount: Int
     # score of perf
     overallPerf: Int #czy wywaliÄ‡?
     weekCount: Int
     goalEffectivenes: Int
 }
