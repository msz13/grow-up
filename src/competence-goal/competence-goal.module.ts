import { Module } from '@nestjs/common';
import { CompetenceGoal } from './models/competence-goal.model';
import { CompetenceGoalService } from './services/competence-goal.service';
import { CompetenceGoalResolvers} from './competence-goal.resolver';
import { TypegooseModule } from 'nestjs-typegoose';
import { DateScalar } from '../common/scalars/date.scalar';
import { ObjectIdScalar } from '../common/scalars/objectId';

@Module({
    imports: [TypegooseModule.forFeature(CompetenceGoal)],
    providers: [CompetenceGoalService, CompetenceGoalResolvers]
})
export class CompetenceGoalModule {}
