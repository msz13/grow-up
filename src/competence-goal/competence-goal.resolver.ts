import { Args, Mutation, Query, Resolver, ResolveProperty, Parent } from '@nestjs/graphql';
import { GoalStatus } from '../common/graphql.schema';
import { CompetenceGoalService } from './services/competence-goal.service'; 
import { CreateCompetenceGoalInput } from './DTO/competence-goal-input';
import { CompetenceGoal, } from './models/competence-goal.model';

import moment = require('moment');
import { UpdateCompetenceGoalInput } from './DTO/competence-goal-update-input';
import format = require('date-fns/format');




@Resolver('CompetenceGoal')
export class CompetenceGoalResolvers {
  constructor(private readonly competenceGoalService: CompetenceGoalService) {}

  @Query()
  competenceGoals(@Args("status") status: GoalStatus, @Args("competence") competence: string): Promise<CompetenceGoal[]>|CompetenceGoal{
    
    if (status===GoalStatus.ACTIVE) {
      
      return  this.competenceGoalService.findActive(competence);
      
    } else 
        return null;
  }

  @Mutation()
  createCompGoal(@Args('competenceGoalInput') args: CreateCompetenceGoalInput): Promise<CompetenceGoal> {
   
    return this.competenceGoalService.create(args);
  } 

  //@Mutation()
  //incPerf(@Args("compGoal_Id") comGoal_Id: string, @Args("fDayOfWeek") fDayOfWeek: number,@Args("dayOfWeek") dayOfWeek: number) {
  //    return this.competenceGoalService.incPerf(comGoal_Id, fDayOfWeek, dayOfWeek, 1)
  //}

  //@Mutation()
  //decPerf (@Args("compGoal_Id") comGoal_Id: string, @Args("fDayOfWeek") fDayOfWeek: number,@Args("dayOfWeek") dayOfWeek: number) {
  //  return this.competenceGoalService.incPerf(comGoal_Id, fDayOfWeek, dayOfWeek, -1)
 // }

  //@Mutation()
  //goalStatusToHold(@Args("goal_Id") id: string) {
  //     return this.competenceGoalService.statusToHold(id);    
     
  //}

  //@Mutation()
  //goalStatusToActive(@Args("goal_Id") id: string) {
  //  return this.competenceGoalService.statusToActive(id);    
 
  //}

  //@Mutation()
  //goalStatusToDone (@Args("goal_Id") id: string){
  //  return this.competenceGoalService.statusToDone(id);
  //}

  //@Mutation()
  //deleteCompGoal(@Args("goal_Id") id: string) {
  //  return this.competenceGoalService.delete(id);
  //}

  //@Mutation()
  //updateComGoal(@Args("goal_Id") id: string, @Args('updateCompetenceGoalInput') args: UpdateCompetenceGoalInput) {
  // 
  //    return this.competenceGoalService.update(id, args);
  //}
  

  //@ResolvePrperty()
  //goalWeekPerf(@Parent() competenceGoal) {
       
  //  return competenceGoal.goalWeekPerf[0];
 // }

 // @ResolveProperty("dayGoalPerf")
 // dayGoalPerf (@Parent() competenceGoal, @Args("dayOfWeek") dayOfWeek: number ){
 //  const dayGoalPerf = competenceGoal[0].goalWeekPerf.daysGaolPerf[dayOfWeek];
 //   return dayGoalPerf;
 // } 

  
  //@ResolveProperty()
  //dayCount(@Parent() competenceGoal: CompetenceGoal, @Args('day') day: Date) {
  
  //return competenceGoal.dayCount(format(day, 'YYYY-MM-DD'));
  //}

  //@ResolveProperty()
  //weekCount(@Parent() competenceGoal: CompetenceGoal, @Args('day') day: Date) {
  // return  competenceGoal.weekCount(day);
       
  //}

  
   
}




