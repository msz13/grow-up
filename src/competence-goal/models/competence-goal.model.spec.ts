import * as mongoose from 'mongoose';
import {CompetenceGoal} from './competence-goal.model'
import { GoalStatus } from '../../common/graphql.schema';
import parse = require('date-fns/parse');
import * as fs from 'fs';
import * as util from 'util';


describe('CompetenceGoal Model Test', () => {

    beforeAll(()=> {
        mongoose.connect('mongodb://test_admin:admin1@ds046037.mlab.com:46037/type_test');
        
        });

    afterAll(()=> {
        mongoose.connection.close();
    });

    beforeEach(()=> {
       
    });
    
    

        test("should save Competence Goal", async ()=>{
          const CompGoalModel = new CompetenceGoal().getModelForClass(CompetenceGoal);
          const compGoal = {
              name: 'mówić dzień dobry brudnej babie',
              competence: "uprzejmość",
              status: GoalStatus.ACTIVE,
              target: 1        
          }

          const  result = {
            name: 'mówić dzień dobry brudnej babie',
            competence: "uprzejmość",
            status: GoalStatus.ACTIVE,
            target: 1

          }
           const competenceGoal = new CompGoalModel(compGoal);
           const foundcompGoal =  await competenceGoal.save();
          
           expect(foundcompGoal).toHaveProperty('competence')
           expect(foundcompGoal).toHaveProperty('target', 1)
           

        })

        test('Competence Goal saveActive', async ()=>{
            const CompGoalModel = new CompetenceGoal().getModelForClass(CompetenceGoal);
            
            const compGoal = {
                name: 'mówić dzień dobry brudnej babie',
                competence: "uprzejmość",
                target: 1        
            }

            let now = parse('2019-01-31');

            const competenceGoal = new CompGoalModel(compGoal);
            
            const foundcompGoal = await competenceGoal.saveActive(now);
            expect(foundcompGoal).not.toBeNull()
            expect(foundcompGoal).toHaveProperty('status', GoalStatus.ACTIVE)
            const activeGoalPerf = foundcompGoal.performance;
            expect(activeGoalPerf.startActive).toEqual(now);
            expect(activeGoalPerf.overallPerf).toBe(0);
            expect(activeGoalPerf.goalPerfEffectivenes).toBe(0);
            expect(activeGoalPerf.dayCount(now)).toBe(1);
            expect(activeGoalPerf.weekCount(now)).toBe(1);
            })
    
    
});