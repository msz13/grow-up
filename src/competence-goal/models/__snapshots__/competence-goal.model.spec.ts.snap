// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetenceGoal Model Test Competence Goal saveActive 1`] = `
Object {
  "__v": 0,
  "_id": "5c5374d8f50c288058899c36",
  "competence": "uprzejmość",
  "name": "mówić dzień dobry brudnej babie",
  "perfHistory": Array [],
  "performance": Object {
    "_id": "5c5374d8f50c288058899c37",
    "_whenToAddDayPerf": 2019-02-06T23:00:00.000Z,
    "goalDayPerf": Array [
      GoalDayPerf {
        "date": 2019-01-30T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
      GoalDayPerf {
        "date": 2019-01-31T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
      GoalDayPerf {
        "date": 2019-02-01T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
      GoalDayPerf {
        "date": 2019-02-02T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
      GoalDayPerf {
        "date": 2019-02-03T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
      GoalDayPerf {
        "date": 2019-02-04T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
      GoalDayPerf {
        "date": 2019-02-05T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
      GoalDayPerf {
        "date": 2019-02-06T23:00:00.000Z,
        "perfCount": 0,
        "targetIsDone": false,
      },
    ],
  },
  "status": "ACTIVE",
  "target": 1,
}
`;
